src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,280 @@
+            readFileToString = FileUtils.readFileToString(new File(SUBSCRIPTION_WITH_JSON_PATH), "UTF-8");
+            readFileToString = stringReplaceText(readFileToString);
+            readSubscriptionNames(readFileToString);
+            postSubscriptions(readFileToString, endPoint);
+            validateSubscriptionsSuccessfullyAdded(endPoint);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        } 
+    }
+    
+    @Then("^Mail subscriptions were triggered$")
+    public void check_mail_subscriptions_were_triggered() throws Throwable {
+        LOGGER.debug("Verifying received emails.");
+        List<SmtpMessage> emails = smtpServer.getReceivedEmails();
+        assert(emails.size() > 0);
+        
+        for(SmtpMessage email : emails) {
+            // assert correct sender.
+            assertEquals(email.getHeaderValue("From"), sender);
+            // assert given test case exist in body.
+            assert(email.getBody().toString().contains("TC5"));
+        }
+    }
+    
+    @Then("^Rest subscriptions were triggered$")
+    public void check_rest_subscriptions_were_triggered() throws Throwable {
+        LOGGER.debug("Verifying REST requests.");
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_PARAMS), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH_PARAMS), Format.JSON))));
+    }
+
+    @When("^I send Eiffel events$")
+    public void send_eiffel_events() {
+        LOGGER.debug("About to send Eiffel events.");
+        try {
+            List<String> eventNames = getEventNamesToSend();
+            JsonNode parsedJSON = getJSONFromFile(EIFFEL_EVENTS_JSON_PATH);
+            int eventsCount = 0;
+
+            for (String eventName : eventNames) {
+                eventsCount++;
+                JsonNode eventJson = parsedJSON.get(eventName);
+                String event = eventJson.toString();
+                rmqHandler.publishObjectToWaitlistQueue(event);
+            }
+
+            assert(waitForEventsToBeProcessed(eventsCount));
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        LOGGER.debug("Eiffel events sent.");
+    }
+
+    private void readSubscriptionNames(String readFileToString) {
+        try {
+            JSONArray array = new JSONArray(readFileToString);
+            for (int i = 0; i < array.length(); i++) {
+                subscriptionNames.add(array.getJSONObject(i).get("subscriptionName").toString());
+            }
```

Comment:
I think that some variable names could be changed to improve the readability a bit. 

What is readFileToString? I dont think that it explains what kind of information it contains(is it a filepath? Is it a jsonstring that previously was read from a file?), possibly what it will be used for at some point.

What kind of information does "array" contain? It gets clearer when we get to array.getJSONObject(i).get("subscriptionName") but without reading that line it could be hard to know. I think it could be clearer.

There are more places with unclear variable names that could be improved, atleast in my mind :)